### PR TITLE
Use puma as prod server as well

### PR DESCRIPTION
**Pull request by segiddins**
Puma seems to be the "standard" nowadays anyway. 

Will monitor https://app.datadoghq.com/dashboard/8t5-9s6-w9q/application-resource-usage?from_ts=1676823387293&to_ts=1676826987293&live=true when deploying to staging

### CODE REVIEW

**Review by indirect — APPROVED** [review]

File: config/puma.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Specifies the `port` that Puma will listen on to receive requests; default is 3000.
+port ENV.fetch("PORT", 3000)
+
+# Specifies the `environment` that Puma will run in.
+environment rails_env
+
+# Specifies the `pidfile` that Puma will use.
+pidfile ENV.fetch("PIDFILE") { "tmp/pids/server.pid" }
+
+preload_app! false
```
Comment (simi) [inline config/puma.rb:42]: Any reason to specify this explicitly? I tried to to compare this to Rails standard puma file and this is not present in there.

**Comment by simi** [comment]
1. What about to rename `unicorn.yaml.erb` as well? Maybe to something generic like `web.yaml.erb` to match `jobs.yaml.erb`?
2. To keep it on super safe side, we can match the current setup of unicorn => 1 thread and 8 workers. This could be configured with `RAILS_MAX_THREADS=1;RAILS_MIN_THREADS=1;WEB_CONCURRENCY=8` if I understand it well.

File: config/puma.rb (hunk 1 above)
Comment (segiddins) [reply config/puma.rb:42]: Copied from unicorn config. Happy to try deleting it!

**Comment by segiddins** [comment]
@simi yeah those both sound good, though I would like to eventually use multiple threads. but we can easily do that in a separate PR.

**Comment by simi** [comment]
> @simi yeah those both sound good, though I would like to eventually use multiple threads. but we can easily do that in a separate PR.

If I remember well, each thread can utilise custom DB connection. I'm not sure there will be enough connections for now on DB side including also background jobs and stuff around. That's why am I conservative in here to start with same setup = same amount of DB connections.

**Comment by segiddins** [comment]
@simi switched to a single thread and remove the `preload_app! false`. I can rename to web.yaml.erb and keep `puma` as the container name?

**Review by simi — APPROVED** [review]

**Comment by simi** [comment]
> @simi switched to a single thread and remove the `preload_app! false`. I can rename to web.yaml.erb and keep `puma` as the container name?

Yup, why not. The only problem I do see is unicorn deployment and service will not be probably removed automatically.  We'll see on staging. It should be simple to remove that manually after deploy once per environment. We should check if this migration is graceful. :pray:

**Comment by segiddins** [comment]
I'll deploy this tomorrow when I'm oncall!

**Comment by simi** [comment]
> I'll deploy this tomorrow when I'm oncall!

To staging?

**Comment by segiddins** [comment]
Yes, to staging

**Pull request merged**